### DISCUSSION TITLE
Built a new Z-Axis with a new spindle, and now i get Alarm 2 when trying to probe for tool length

### DISCUSSION THREAD

**Discussion by Klippert** [terjeio/ioSender#336] [Q&A] [answered]
Hello,

I'm somewhat baffled here. I've got a nice CNC that's been working reliably for over 2.5 years now. Then i finally pulled myself together and made a new Z-Axis with better guides, ballscrew and so on. And its working like a charm. On the new Z-Axis i also got room for a new 2.2kw watercooled china spindle and watercooling. So far so good. 
I had some issues controlling the VFD as the inputs gennerally just wants an "open" input instead of a +5v high. But i found my way around it with a little mosfet board. So VFD is working and Spindle control is working. The Z-axis behaves perfectly when jogging. 

As the New Z-axis is higher than the old one, I have updated the Z-Travel to 205mm. (soft limit)
I've also had to increase the probing distance to 170mm, where 80mm was sufficient before.

Now when i probe my tool length sensor, I get this behaviour:

1. I home the machine
2. I go to probing and select all three (Probe Fixture... , Establish reference offset, Set coordinate system)
3. I click start, and the machine lift z to max, moves over the probe and starts the initial probing seek. 
4. When the tool hits the probe it retracts 10mm (as i played with the latch distance to see if that made a difference)
5. Then i get a probing failed. as in the below screenshot:
<img width="684" alt="Screenshot 2023-10-05 211852" src="https://github.com/terjeio/ioSender/assets/53910032/d1e35f6f-f666-468f-873a-851d3d8883b2">

I cant go to the GRBL tab and see the error, though by closing iosender, and then reopening it, it comes up with a message "unable to communicate, trying a soft reset, and then it shows the Alarm 2 state.

Any suggestions would be most welcome, as my logic says that there should be no changes.

Best
Kristoffer

**Comment by terjeio** [comment]
Alarm 2 is soft limits triggered.
To see what is happening open the console via the _File_ menu and check _Verbose_. The last 10-20 lines before the alarm may provide clues.
BTW which version of ioSender and the controller are you using?

> I cant go to the GRBL tab and see the error

Is the tab greyed out or not responding on clicks?

**Comment by Klippert** [comment]
I'll check via the file menu with Verbose.

The tab is not greyed out, it's just not responding to clicks.
Version of IOSender is 2.0.30 - i don't think i updated it since i got
everything working about 2.5 years ago.

Grbl is set up as follows:

 %
; grblHAL
; 1.1f.20210313
; [OPT:VNMPZHSL+,35,1024,4,0]
; [NEWOPT:ENUMS,RT+,HOME,ES,TC,ETH]
; [FIRMWARE:grblHAL]
; [NVS STORAGE:*FLASH]
; [DRIVER:iMXRT1062]
; [DRIVER VERSION:210313]
; [DRIVER OPTIONS:USB.2]
; [BOARD:T41U5XBB]
; [IP:0.0.0.0]
;
$N0=
$N1=
; 0 - Step pulse time
$0=10.0
; 1 - Step idle delay
$1=255
; 2 - Step pulse invert
$2=1
; 3 - Step direction invert
$3=6
; 4 - Invert step enable pin(s)
$4=7
; 5 - Invert limit pins
$5=15
; 6 - Invert probe pin
$6=0
; 7 - Disable spindle with zero speed
$7=0
; 10 - Status report options
$10=511
; 11 - Junction deviation
$11=0.010
; 12 - Arc tolerance
$12=0.002
; 13 - Report in inches
$13=0
; 14 - Invert control pins
$14=79
; 15 - Invert coolant pins
$15=0
; 16 - Invert spindle signals
$16=0
; 17 - Pullup disable control pins
$17=0
; 18 - Pullup disable limit pins
$18=0
; 19 - Pullup disable probe pin
$19=0
; 20 - Soft limits enable
$20=1
; 21 - Hard limits enable
$21=1
; 22 - Homing cycle
$22=11
; 23 - Homing direction invert
$23=3
; 24 - Homing locate feed rate
$24=25.0
; 25 - Homing search seek rate
$25=1800.0
; 26 - Homing switch debounce delay
$26=250
; 27 - Homing switch pull-off distance
$27=1.200
; 28 - G73 Retract distance
$28=0.100
; 29 - Pulse delay
$29=0.0
; 30 - Maximum spindle speed
$30=24000.000
; 31 - Minimum spindle speed
$31=0.000
; 32 - Mode of operation
$32=0
; 33 - Spindle PWM frequency
$33=5000
; 34 - Spindle PWM off value
$34=0.0
; 35 - Spindle PWM min value
$35=0.0
; 36 - Spindle PWM max value
$36=100.0
; 37 - Steppers deenergize
$37=7
; 39 - Enable legacy RT commands
$39=1
; 40 - Limit jog commands
$40=1
; 41 - Parking cycle
$41=1
; 42 - Parking axis
$42=2
; 43 - Homing passes
$43=1
; 44 - Axes homing, first pass
$44=4
; 45 - Axes homing, second pass
$45=3
; 46 - Axes homing, third pass
$46=0
; 47 - Axes homing, fourth pass
$47=0
; 56 - Parking pull-out distance
$56=5.0
; 57 - Parking pull-out rate
$57=100.0
; 58 - Parking target
$58=-5.0
; 59 - Parking fast rate
$59=1000.0
; 60 - Restore overrides
$60=0
; 61 - Safety door options
$61=0
; 62 - Sleep enable
$62=0
; 63 - Feed hold actions
$63=2
; 64 - Force init alarm
$64=0
; 65 - Probing feed override
$65=0
; 70 - Network Services
$70=3
; 80 - Spindle P-gain
$80=1.000
; 81 - Spindle I-gain
$81=0.010
; 82 - Spindle D-gain
$82=0.000
; 84 - Spindle PID max error
$84=0.000
; 85 - Spindle PID max I error
$85=10.000
; 90 - Spindle sync P-gain
$90=0.000
; 91 - Spindle sync I-gain
$91=0.000
; 92 - Spindle sync D-gain
$92=0.000
; 95 - Spindle sync PID max I error
$95=0.000
; 100 - X-axis travel resolution
$100=200.000
; 101 - Y-axis travel resolution
$101=200.000
; 102 - Z-axis travel resolution
$102=500.000
; 103 - A-axis travel resolution
$103=250.000
; 110 - X-axis maximum rate
$110=15000.000
; 111 - Y-axis maximum rate
$111=15000.000
; 112 - Z-axis maximum rate
$112=8000.000
; 113 - A-axis maximum rate
$113=500.000
; 120 - X-axis acceleration
$120=125.000
; 121 - Y-axis acceleration
$121=120.000
; 122 - Z-axis acceleration
$122=75.000
; 123 - A-axis acceleration
$123=1000.000
; 130 - X-axis maximum travel
$130=253.000
; 131 - Y-axis maximum travel
$131=522.000
; 132 - Z-axis maximum travel
$132=205.000
; 133 - A-axis maximum travel
$133=200.000
; 300 - Hostname
$300=GRBL
; 301 - IP Mode
$301=1
; 302 - IP Address
$302=192.168.5.1
; 303 - Gateway
$303=192.168.5.1
; 304 - Netmask
$304=255.255.255.0
; 305 - Telnet port
$305=23
; 307 - Websocket port
$307=80
; 341 - Tool change mode
$341=2
; 342 - Tool change probing distance
$342=170.0
; 343 - Tool change locate feed rate
$343=25.0
; 344 - Tool change search seek rate
$344=350.0
; 345 - Tool change probe pull-off rate
$345=500.0
; 347 - Dual axis length fail
$347=5.0
; 348 - Dual axis length fail min
$348=2.500
; 349 - Dual axis length fail max
$349=25.000
; 370 - Invert I/O Port inputs
$370=0
; 372 - Invert I/O Port outputs
$372=0
%

**Reply by Klippert** [reply]
This is the probing with verbose logging: 

<Idle|MPos:0.000,370.000,0.000,0.000|Bf:35,1023|FS:0,0>
<Idle|MPos:0.000,370.000,0.000,0.000|Bf:35,1023|FS:0,0>
<Idle|MPos:0.000,370.000,0.000,0.000|Bf:35,1023|FS:0,0>
<Idle|MPos:0.000,370.000,0.000,0.000|Bf:35,1023|FS:0,0>
<Idle|MPos:0.000,370.000,0.000,0.000|Bf:35,1023|FS:0,0|WCO:0.000,0.000,0.000,0.000|Ov:100,100,100>
<Idle|MPos:0.000,370.000,0.000,0.000|Bf:35,1023|FS:0,0|WCO:0.000,0.000,0.000,0.000|WCS:G54|Ov:100,100,100|A:|Sc:|H:1,7|T:0|TLR:0|FW:grblHAL>
Probing...
G91F300
ok
PM:ok
G53G0Z0
ok
PM:ok
G53G0X1Y521
ok
PM:ok
G53G0Z0
ok
PM:ok
G38.3F300Z-170
<Run|MPos:0.005,370.780,0.000,0.000|Bf:34,1023|FS:1440,0|WCO:0.000,0.000,0.000,0.000|WCS:G54|Ov:100,100,100|A:|Sc:|H:1,7|T:0|TLR:0|FW:grblHAL>
<Run|MPos:0.040,375.905,0.000,0.000|Bf:34,1023|FS:2880,0>
<Run|MPos:0.105,385.885,0.000,0.000|Bf:34,1023|FS:4320,0>
<Run|MPos:0.210,401.825,0.000,0.000|Bf:34,1023|FS:5832,0>
<Run|MPos:0.345,421.780,0.000,0.000|Bf:34,1023|FS:7272,0>
<Run|MPos:0.505,446.555,0.000,0.000|Bf:34,1023|FS:7441,0>
<Run|MPos:0.670,470.960,0.000,0.000|Bf:34,1023|FS:5929,0>
<Run|MPos:0.800,490.455,0.000,0.000|Bf:34,1023|FS:4489,0>
<Run|MPos:0.900,506.015,0.000,0.000|Bf:34,1023|FS:2977,0>
<Run|MPos:0.965,515.640,0.000,0.000|Bf:34,1023|FS:1537,0>
<Run|MPos:0.995,520.420,0.000,0.000|Bf:34,1023|FS:97,0>
<Run|MPos:1.000,521.000,-0.350,0.000|Bf:34,1023|FS:300,0>
<Run|MPos:1.000,521.000,-1.354,0.000|Bf:34,1023|FS:300,0>
<Run|MPos:1.000,521.000,-2.354,0.000|Bf:34,1023|FS:300,0>
<Run|MPos:1.000,521.000,-3.422,0.000|Bf:34,1023|FS:300,0>
<Run|MPos:1.000,521.000,-4.428,0.000|Bf:34,1023|FS:300,0>
<Run|MPos:1.000,521.000,-5.432,0.000|Bf:34,1023|FS:300,0>
<Run|MPos:1.000,521.000,-6.432,0.000|Bf:34,1023|FS:300,0>
<Run|MPos:1.000,521.000,-7.436,0.000|Bf:34,1023|FS:300,0>
<Run|MPos:1.000,521.000,-8.440,0.000|Bf:34,1023|FS:300,0>
<Run|MPos:1.000,521.000,-9.500,0.000|Bf:34,1023|FS:300,0|Ov:100,100,100>
<Run|MPos:1.000,521.000,-10.504,0.000|Bf:34,1023|FS:300,0>
<Run|MPos:1.000,521.000,-11.508,0.000|Bf:34,1023|FS:300,0>
<Run|MPos:1.000,521.000,-12.514,0.000|Bf:34,1023|FS:300,0>
<Run|MPos:1.000,521.000,-13.518,0.000|Bf:34,1023|FS:300,0>
<Run|MPos:1.000,521.000,-14.578,0.000|Bf:34,1023|FS:300,0>
<Run|MPos:1.000,521.000,-15.602,0.000|Bf:34,1023|FS:300,0>
<Run|MPos:1.000,521.000,-16.586,0.000|Bf:34,1023|FS:300,0>
<Run|MPos:1.000,521.000,-17.586,0.000|Bf:34,1023|FS:300,0>
<Run|MPos:1.000,521.000,-18.590,0.000|Bf:34,1023|FS:300,0>
<Run|MPos:1.000,521.000,-19.654,0.000|Bf:34,1023|FS:300,0|WCO:0.000,0.000,0.000,0.000>
<Run|MPos:1.000,521.000,-20.660,0.000|Bf:34,1023|FS:300,0>
<Run|MPos:1.000,521.000,-21.664,0.000|Bf:34,1023|FS:300,0>
<Run|MPos:1.000,521.000,-22.664,0.000|Bf:34,1023|FS:300,0>
<Run|MPos:1.000,521.000,-23.668,0.000|Bf:34,1023|FS:300,0>
<Run|MPos:1.000,521.000,-24.732,0.000|Bf:34,1023|FS:300,0>
<Run|MPos:1.000,521.000,-25.732,0.000|Bf:34,1023|FS:300,0>
<Run|MPos:1.000,521.000,-26.736,0.000|Bf:34,1023|FS:300,0>
<Run|MPos:1.000,521.000,-27.736,0.000|Bf:34,1023|FS:300,0>
<Run|MPos:1.000,521.000,-28.740,0.000|Bf:34,1023|FS:300,0>
<Run|MPos:1.000,521.000,-29.810,0.000|Bf:34,1023|FS:300,0|Ov:100,100,100>
<Run|MPos:1.000,521.000,-30.808,0.000|Bf:34,1023|FS:300,0>
<Run|MPos:1.000,521.000,-31.814,0.000|Bf:34,1023|FS:300,0>
<Run|MPos:1.000,521.000,-32.818,0.000|Bf:34,1023|FS:300,0>
<Run|MPos:1.000,521.000,-33.818,0.000|Bf:34,1023|FS:300,0>
<Run|MPos:1.000,521.000,-34.886,0.000|Bf:34,1023|FS:300,0>
<Run|MPos:1.000,521.000,-35.892,0.000|Bf:34,1023|FS:300,0>
<Run|MPos:1.000,521.000,-36.896,0.000|Bf:34,1023|FS:300,0>
<Run|MPos:1.000,521.000,-37.896,0.000|Bf:34,1023|FS:300,0>
<Run|MPos:1.000,521.000,-38.904,0.000|Bf:34,1023|FS:300,0>
<Run|MPos:1.000,521.000,-39.910,0.000|Bf:34,1023|FS:300,0>
<Run|MPos:1.000,521.000,-40.974,0.000|Bf:34,1023|FS:300,0>
<Run|MPos:1.000,521.000,-41.974,0.000|Bf:34,1023|FS:300,0>
<Run|MPos:1.000,521.000,-42.978,0.000|Bf:34,1023|FS:300,0>
<Run|MPos:1.000,521.000,-43.978,0.000|Bf:34,1023|FS:300,0>
<Run|MPos:1.000,521.000,-44.982,0.000|Bf:34,1023|FS:300,0>
<Run|MPos:1.000,521.000,-46.052,0.000|Bf:34,1023|FS:300,0>
<Run|MPos:1.000,521.000,-47.056,0.000|Bf:34,1023|FS:300,0>
<Run|MPos:1.000,521.000,-48.056,0.000|Bf:34,1023|FS:300,0>
<Run|MPos:1.000,521.000,-49.060,0.000|Bf:34,1023|FS:300,0>
<Run|MPos:1.000,521.000,-50.060,0.000|Bf:34,1023|FS:300,0|WCO:0.000,0.000,0.000,0.000|Ov:100,100,100>
<Run|MPos:1.000,521.000,-51.130,0.000|Bf:34,1023|FS:300,0>
<Run|MPos:1.000,521.000,-52.134,0.000|Bf:34,1023|FS:300,0>
<Run|MPos:1.000,521.000,-53.134,0.000|Bf:34,1023|FS:300,0>
<Run|MPos:1.000,521.000,-54.138,0.000|Bf:34,1023|FS:300,0>
<Run|MPos:1.000,521.000,-55.138,0.000|Bf:34,1023|FS:300,0>
<Run|MPos:1.000,521.000,-56.202,0.000|Bf:34,1023|FS:300,0>
<Run|MPos:1.000,521.000,-57.208,0.000|Bf:34,1023|FS:300,0>
<Run|MPos:1.000,521.000,-58.206,0.000|Bf:34,1023|FS:300,0>
<Run|MPos:1.000,521.000,-59.212,0.000|Bf:34,1023|FS:300,0>
<Run|MPos:1.000,521.000,-60.216,0.000|Bf:34,1023|FS:300,0>
<Run|MPos:1.000,521.000,-61.286,0.000|Bf:34,1023|FS:300,0>
<Run|MPos:1.000,521.000,-62.290,0.000|Bf:34,1023|FS:300,0>
<Run|MPos:1.000,521.000,-63.290,0.000|Bf:34,1023|FS:300,0>
<Run|MPos:1.000,521.000,-64.290,0.000|Bf:34,1023|FS:300,0>
<Run|MPos:1.000,521.000,-65.294,0.000|Bf:34,1023|FS:300,0>
<Run|MPos:1.000,521.000,-66.358,0.000|Bf:34,1023|FS:300,0>
<Run|MPos:1.000,521.000,-67.358,0.000|Bf:34,1023|FS:300,0>
<Run|MPos:1.000,521.000,-68.362,0.000|Bf:34,1023|FS:300,0>
<Run|MPos:1.000,521.000,-69.368,0.000|Bf:34,1023|FS:300,0>
<Run|MPos:1.000,521.000,-70.366,0.000|Bf:34,1023|FS:300,0|Ov:100,100,100>
<Run|MPos:1.000,521.000,-71.432,0.000|Bf:34,1023|FS:300,0>
<Run|MPos:1.000,521.000,-72.430,0.000|Bf:34,1023|FS:300,0>
<Run|MPos:1.000,521.000,-73.436,0.000|Bf:34,1023|FS:300,0>
<Run|MPos:1.000,521.000,-74.440,0.000|Bf:34,1023|FS:300,0>
<Run|MPos:1.000,521.000,-75.440,0.000|Bf:34,1023|FS:300,0>
<Run|MPos:1.000,521.000,-76.510,0.000|Bf:34,1023|FS:300,0>
<Run|MPos:1.000,521.000,-77.514,0.000|Bf:34,1023|FS:300,0>
<Run|MPos:1.000,521.000,-78.518,0.000|Bf:34,1023|FS:300,0>
<Run|MPos:1.000,521.000,-79.518,0.000|Bf:34,1023|FS:300,0>
<Run|MPos:1.000,521.000,-80.522,0.000|Bf:34,1023|FS:300,0|WCO:0.000,0.000,0.000,0.000>
<Run|MPos:1.000,521.000,-81.586,0.000|Bf:34,1023|FS:300,0>
<Run|MPos:1.000,521.000,-82.592,0.000|Bf:34,1023|FS:300,0>
<Run|MPos:1.000,521.000,-83.596,0.000|Bf:34,1023|FS:300,0>
<Run|MPos:1.000,521.000,-84.596,0.000|Bf:34,1023|FS:300,0>
<Run|MPos:1.000,521.000,-85.600,0.000|Bf:34,1023|FS:300,0>
<Run|MPos:1.000,521.000,-86.670,0.000|Bf:34,1023|FS:300,0>
<Run|MPos:1.000,521.000,-87.670,0.000|Bf:34,1023|FS:300,0>
<Run|MPos:1.000,521.000,-88.668,0.000|Bf:34,1023|FS:300,0>
<Run|MPos:1.000,521.000,-89.674,0.000|Bf:34,1023|FS:300,0>
<Run|MPos:1.000,521.000,-90.674,0.000|Bf:34,1023|FS:300,0|Ov:100,100,100>
<Run|MPos:1.000,521.000,-91.742,0.000|Bf:34,1023|FS:300,0>
<Run|MPos:1.000,521.000,-92.748,0.000|Bf:34,1023|FS:300,0>
<Run|MPos:1.000,521.000,-93.746,0.000|Bf:34,1023|FS:300,0>
<Run|MPos:1.000,521.000,-94.752,0.000|Bf:34,1023|FS:300,0>
<Run|MPos:1.000,521.000,-95.750,0.000|Bf:34,1023|FS:300,0>
<Run|MPos:1.000,521.000,-96.816,0.000|Bf:34,1023|FS:300,0>
<Run|MPos:1.000,521.000,-97.816,0.000|Bf:34,1023|FS:300,0>
<Run|MPos:1.000,521.000,-98.820,0.000|Bf:34,1023|FS:300,0>
<Run|MPos:1.000,521.000,-99.824,0.000|Bf:34,1023|FS:300,0>
<Run|MPos:1.000,521.000,-100.824,0.000|Bf:34,1023|FS:300,0>
<Run|MPos:1.000,521.000,-101.894,0.000|Bf:34,1023|FS:300,0>
<Run|MPos:1.000,521.000,-102.892,0.000|Bf:34,1023|FS:300,0>
<Run|MPos:1.000,521.000,-103.898,0.000|Bf:34,1023|FS:300,0>
<Run|MPos:1.000,521.000,-104.898,0.000|Bf:34,1023|FS:300,0>
<Run|MPos:1.000,521.000,-105.902,0.000|Bf:34,1023|FS:300,0>
<Run|MPos:1.000,521.000,-106.970,0.000|Bf:34,1023|FS:300,0>
<Run|MPos:1.000,521.000,-107.976,0.000|Bf:34,1023|FS:300,0>
<Run|MPos:1.000,521.000,-108.974,0.000|Bf:34,1023|FS:300,0>
<Run|MPos:1.000,521.000,-109.980,0.000|Bf:34,1023|FS:300,0>
<Run|MPos:1.000,521.000,-110.984,0.000|Bf:34,1023|FS:300,0|WCO:0.000,0.000,0.000,0.000|Ov:100,100,100>
<Run|MPos:1.000,521.000,-111.894,0.000|Bf:34,1023|FS:0,0|Pn:P>
[PRB:1.000,521.000,-111.326,0.000:1]
ok
PM:ok
G0Z10
ok
PM:ok
G38.3F25Z-170
<Run|MPos:1.000,521.000,-111.842,0.000|Bf:34,1023|FS:585,0|Pn:P>
<Run|MPos:1.000,521.000,-109.734,0.000|Bf:34,1023|FS:1485,0>
<Run|MPos:1.000,521.000,-105.056,0.000|Bf:34,1023|FS:901,0>
<Run|MPos:1.000,521.000,-102.204,0.000|Bf:34,1023|FS:1,0>
ALARM:2
PM:fail
G90
Probing cancelled/failed (ALARM)
$TLR
$G
Probing failed
[MSG:Reset to continue]
<Alarm|MPos:1.000,521.000,-101.906,0.000|Bf:35,1011|FS:0,0>
<Alarm|MPos:1.000,521.000,-101.906,0.000|Bf:35,1011|FS:0,0>
<Alarm|MPos:1.000,521.000,-101.906,0.000|Bf:35,1011|FS:0,0>
<Alarm|MPos:1.000,521.000,-101.906,0.000|Bf:35,1011|FS:0,0>
<Alarm|MPos:1.000,521.000,-101.906,0.000|Bf:35,1011|FS:0,0>
<Alarm|MPos:1.000,521.000,-101.906,0.000|Bf:35,1011|FS:0,0>
<Alarm|MPos:1.000,521.000,-101.906,0.000|Bf:35,1011|FS:0,0>
<Alarm|MPos:1.000,521.000,-101.906,0.000|Bf:35,1011|FS:0,0>
<Alarm|MPos:1.000,521.000,-101.906,0.000|Bf:35,1011|FS:0,0>
<Alarm|MPos:1.000,521.000,-101.906,0.000|Bf:35,1011|FS:0,0>
<Alarm|MPos:1.000,521.000,-101.906,0.000|Bf:35,1011|FS:0,0>
<Alarm|MPos:1.000,521.000,-101.906,0.000|Bf:35,1011|FS:0,0>
<Alarm|MPos:1.000,521.000,-101.906,0.000|Bf:35,1011|FS:0,0>
<Alarm|MPos:1.000,521.000,-101.906,0.000|Bf:35,1011|FS:0,0>
<Alarm|MPos:1.000,521.000,-101.906,0.000|Bf:35,1011|FS:0,0|Ov:100,100,100>
<Alarm|MPos:1.000,521.000,-101.906,0.000|Bf:35,1011|FS:0,0>
<Alarm|MPos:1.000,521.000,-101.906,0.000|Bf:35,1011|FS:0,0>
<Alarm|MPos:1.000,521.000,-101.906,0.000|Bf:35,1011|FS:0,0>

**Reply by phil-barrett** [reply]
You really should update your firmware.  The good news is the[ web builder](http://svn.io-engineering.com:8080/) makes it easy to stay current.

**Reply by terjeio** [reply, marked as answer]
And update the sender as well since the version you do not use the latch distance for the second probe move, later versions uses the latch distance * 1.5, minimum 2mm, to avoid triggering soft limits.:

PM:ok
**G38.3F25Z-170** _This is triggering soft limits._
<Run|MPos:1.000,521.000,-111.842,0.000|Bf:34,1023|FS:585,0|Pn:P>
<Run|MPos:1.000,521.000,-109.734,0.000|Bf:34,1023|FS:1485,0>
<Run|MPos:1.000,521.000,-105.056,0.000|Bf:34,1023|FS:901,0>
<Run|MPos:1.000,521.000,-102.204,0.000|Bf:34,1023|FS:1,0>
ALARM:2
PM:fail

**Reply by Klippert** [reply]
Hello Again, 

Thanks for the suggestions. 
I just increased the Z-Axis max length to 500mm and now probing works. This kills the softlimit on Z but at least I was able to run my Job. So this confirms what was wrong.

Now in an attempt to build a new firmware I went through the web interface, and of course I can't remember half the configuration from 2 years ago when I last built it :)
So it's a bit of a case of from the top.
Is there an easy-ish way of digging out my old configuration files and checking my options are the same?

**Reply by terjeio** [reply]
It seems to be a basic build with dual axis (Y auto squared?) and safety door input enabled. From the settings dump above:
```
; [DRIVER:iMXRT1062]
...
; [BOARD:T41U5XBB]
```
Ethernet is enabled but since the IP address is not set it is not in use.

Backup before and restore after flashing the controller settings from the _Settings: Grbl_ tab.
